### PR TITLE
api: add percentile node usage to NodeMetric

### PR DESCRIPTION
**Pull request by zwzhang0107**
Signed-off-by: 佑祎 <zzw261520@alibaba-inc.com>

### Ⅰ. Describe what this PR does
add percentile usage info of node for scheduler/de-scheduler to solve the load-aware issue.

### Ⅱ. Does this pull request fix one issue?
#856

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: apis/slo/v1alpha1/nodemetric_types.go [hunk 1]
```diff
@@ -23,8 +23,25 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type AggregationType string
+
+const (
+	// max is not welcomed since it may import outliers
```
Comment (jasonliu747) [inline apis/slo/v1alpha1/nodemetric_types.go:29]: do we have to mention `max` here?

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

File: apis/slo/v1alpha1/nodemetric_types.go (hunk 1 above)
Comment (zwzhang0107) [reply apis/slo/v1alpha1/nodemetric_types.go:29]: in order to explain the reason that why we don't add the `MAX` aggregate type temporarily.

**Review by eahydra — COMMENTED** [review]
/lgtm

File: apis/slo/v1alpha1/nodemetric_types.go [hunk 2]
```diff
@@ -23,8 +23,25 @@ import (
...
+	AVG    AggregationType = "avg"
+	P99    AggregationType = "p99"
+	P95    AggregationType = "p95"
+	P90    AggregationType = "p90"
+	MEDIAN AggregationType = "median"
+)
+
 type NodeMetricInfo struct {
-	NodeUsage ResourceMap `json:"nodeUsage,omitempty"`
+	NodeUsage            ResourceMap       `json:"nodeUsage,omitempty"`
```
Comment (FillZpp) [inline apis/slo/v1alpha1/nodemetric_types.go:38]: If the field is optional, it had better be a pointer type like `*ResourceMap`, or we can remove the `omitempty` to make it a required field.

File: apis/slo/v1alpha1/nodemetric_types.go [hunk 3]
```diff
@@ -23,8 +23,25 @@ import (
...
 
+type AggregationType string
+
+const (
+	// max is not welcomed since it may import outliers
+	AVG    AggregationType = "avg"
+	P99    AggregationType = "p99"
+	P95    AggregationType = "p95"
+	P90    AggregationType = "p90"
+	MEDIAN AggregationType = "median"
```
Comment (hormes) [inline apis/slo/v1alpha1/nodemetric_types.go:34]: Why not use p50 to keep consistently

Comment (eahydra) [reply apis/slo/v1alpha1/nodemetric_types.go:34]: Maybe in the percentile statistical method, the median is generally used to represent p50

File: apis/slo/v1alpha1/nodemetric_types.go (hunk 2 above)
Comment (zwzhang0107) [reply apis/slo/v1alpha1/nodemetric_types.go:38]: it is an internal protocol, and ResourceMap contains an inline map. so it does not need to be defined as pointer type.

**Review by FillZpp — COMMENTED** [review]
/lgtm

**Pull request merged**